index/index.go

```diff
@@ -951,25 +951,28 @@ func (r *Reader) LabelNames() ([]string, error) {
 type stringTuples struct {
 	length  int      // tuple length
 	entries []string // flattened tuple entries
+	swapBuf []string
 }
 
 func NewStringTuples(entries []string, length int) (*stringTuples, error) {
 	if len(entries)%length != 0 {
 		return nil, errors.Wrap(encoding.ErrInvalidSize, "string tuple list")
 	}
-	return &stringTuples{entries: entries, length: length}, nil
+	return &stringTuples{
+		entries: entries,
+		length:  length,
+		swapBuf: make([]string, length),
```

Comment:
Would it be worthwhile to lazily initialize the swap buffer? How often are string tuples created and `Swap` never called?

Reply:
During compaction it would always call `Swap` (via the `Sort` method) https://github.com/prometheus/tsdb/blob/d230c67aa180850b80ae49e07079f55df1da0502/index/index.go#L412-L416

But the one returned by head (used in `querier.go`) does not call sort `Sort` on it. https://github.com/prometheus/tsdb/blob/d230c67aa180850b80ae49e07079f55df1da0502/head.go#L1230

I will make it lazy. I was just worried about lot of redundant checks in the `Swap`, but I guess it won't hurt perf.